test/transcoding/OpenCL/atomic_syncscope_test.ll

```diff
@@ -0,0 +1,157 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_EXT_shader_atomic_float_add -o %t.spv
+; RUN: spirv-val %t.spv
+; RUN: llvm-spirv %t.spv -to-text -o %t.spt
+; RUN: FileCheck < %t.spt %s -check-prefix=CHECK-SPIRV
+; RUN: llvm-spirv %t.spv -r --spirv-target-env=CL2.0 -o - | llvm-dis -o %t.rev.ll
+; RUN: FileCheck < %t.rev.ll %s -check-prefix=CHECK-LLVM
+
+target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir64"
+
+@j = local_unnamed_addr addrspace(1) global i32 0, align 4
+
+; SPIRV scopes
+; 0 - ScopeCrossDevice (all svm devices)
+; 1 - ScopeDevice
+; 2 - ScopeWorkGroup
+; 3 - ScopeSubgroup
+; 4 - ScopeInvocation (mapped from work item)
+
+; OCL scopes
+; 0 - work_item
+; 1 - work_group
+; 2 - device
+; 3 - all_svm_devices
+; 4 - sub_group
+
+; CHECK-SPIRV-DAG: Constant [[#]] [[#ConstInt0:]] 0
+; CHECK-SPIRV-DAG: Constant [[#]] [[#SequentiallyConsistent:]] 16
+; CHECK-SPIRV-DAG: Constant [[#]] [[#ConstInt1:]] 1
+; CHECK-SPIRV-DAG: Constant [[#]] [[#ConstInt2:]] 2
+; CHECK-SPIRV-DAG: Constant [[#]] [[#ConstInt3:]] 3
+; CHECK-SPIRV-DAG: Constant [[#]] [[#Const2Power30:]] 1073741824 
+; CHECK-SPIRV-DAG: Constant [[#]] [[#ConstInt42:]] 42
+
+; AtomicLoad ResTypeId ResId PtrId MemScopeId MemSemanticsId
+; CHECK-SPIRV: AtomicLoad [[#]] [[#]] [[#]] [[#ConstInt2]] [[#SequentiallyConsistent]]
+; CHECK-SPIRV: AtomicLoad [[#]] [[#]] [[#]] [[#ConstInt1]] [[#SequentiallyConsistent]]
+; CHECK-SPIRV: AtomicLoad [[#]] [[#]] [[#]] [[#ConstInt1]] [[#SequentiallyConsistent]]
+; CHECK-SPIRV: AtomicLoad [[#]] [[#]] [[#]] [[#ConstInt3]] [[#SequentiallyConsistent]]
+
+; CHECK-LLVM: call spir_func i32 @_Z20atomic_load_explicitPU3AS4VU7_Atomici12memory_order12memory_scope(ptr{{.*}}, i32 5, i32 1)
+; CHECK-LLVM: call spir_func i32 @_Z20atomic_load_explicitPU3AS4VU7_Atomici12memory_order12memory_scope(ptr{{.*}}, i32 5, i32 2)
+; CHECK-LLVM: call spir_func i32 @_Z20atomic_load_explicitPU3AS4VU7_Atomici12memory_order12memory_scope(ptr{{.*}}, i32 5, i32 2)
+; CHECK-LLVM: call spir_func i32 @_Z20atomic_load_explicitPU3AS4VU7_Atomici12memory_order12memory_scope(ptr{{.*}}, i32 5, i32 4)
+
+define dso_local void @fi1(ptr addrspace(4) nocapture noundef readonly %i) local_unnamed_addr #0 {
+entry:
+  %0 = load atomic i32, ptr addrspace(4) %i syncscope("workgroup") seq_cst, align 4
+  %1 = load atomic i32, ptr addrspace(4) %i syncscope("device") seq_cst, align 4
+  %2 = load atomic i32, ptr addrspace(4) %i seq_cst, align 4
```

Comment:
Funny question, what would happen if we pass a non-empty string, that doesn't match the table? I'd hope that we will fall to default (device?), right?

Reply:
I'm afraid it'll fail in the current implementation with "not found in map" error. Isn't it a correct behavior, or do we expect some graceful handling like setting the default value instead?

Reply:
What is the current (without the patch) behavior, when sync scope is passed? I believe it should remain for the unknown scopes.

Reply:
Fixed to preserve the current behavior, and now we don't need the default value for empty string scope `""` (but we still may have one though, if it's different from device)